base/src/main/java/org/mozilla/jss/ssl/javax/JSSEngineReferenceImpl.java

```diff
@@ -366,13 +370,17 @@ private void createBufferFD() throws SSLException {
     private void initClient() throws SSLException {
         debug("JSSEngine: initClient()");
 
-        if (cert != null && key != null) {
+        if (certs != null && !certs.isEmpty()) {
             // NSS uses a callback to check for the client certificate; we
             // assume we have knowledge of it ahead of time and set it
             // directly on our SSLFDProxy instance.
             //
             // In the future, we could use a KeyManager for inquiring at
             // selection time which certificate to use.
+            
+            // Clients do not support multiple certificates so only the first
+            // is used.
+            PK11Cert cert = certs.iterator().next().getLeft();
```

Comment:
If this is a callback, then the server should have sent a list of CA's that are relevent. You should select the cert based on the CA and the negotiated auth value. So it's not quite right to say the client only supports one cert, it's at this point it narrows is selection of a cert to one. For your usage, this is probably fine to use the first one, but the client's support of multiple certs depends on the callback's implementation.